### ISSUE TITLE
Login slowness and a lot of correlations

### ISSUE THREAD

**Issue by Tommi852** [MISP/MISP#7396]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Support
| OS version (server&client)       | Ubuntu 18.04.4 LTS x86_64
| MISP version / git hash   |  v2.4.142 (81c2d01bcdeb859aa867c16bd7169c5824d93e46) 
| Browser                   | Firefox & Chromium

#### In short:
I have performance issues on my MISP instance.
I guess my question in short is, if it is normal to have 150 million correlations for 1,3 million attributes as it seems to be the source of my performance issues? See the screenshot.

#### Longer description:
I'm experiencing performance issues on my MISP istance. For example login often takes 2-5mins to actually process. I can see the login event on syslog right away when I submit the credentials, but the page still keeps loading for the ~5 minutes.
Also opening the statistics page takes few minutes. There's also general slowness of opening other pages.

Looking at htop I can see two out of the four CPU cores peaking at 100% every 10 seconds.
Looks like php ./bin/resque has CPU% of 100% almost constantly.

I found similiar issue at https://github.com/MISP/MISP/issues/3851, though all five of my CSV feeds are already going to single fixed events and rest are MISP format feeds.
I also tried the solution on that issue by disabling my correlations and building them again.
I regained performance briefly after the correlations were still building, but now they're back to over 150 million correlations and the instance is slow again.

### Logs, screenshots, configuration dump, ...
![MISP 1 statistics 06052021](https://user-images.githubusercontent.com/31437167/117286678-27131980-ae72-11eb-8266-116874837fc6.PNG)

**Comment by iglocska** [comment]
That attribute : correlation ratio is indeed pretty crazy. Sounds like one of your data-sources is replicating the same data set over and over...

**Comment by jloehel** [comment]
@iglocska We have similar scenario with ~185.000.000 correlations and ~8.500.000 attributes. I see a high load on the mysql server this causes sometimes hiccups with the login, displaying events or generating reports. Is there any best practice to cleanup correlations? I have already added Correlation Exclusions. Thanks for this nice feature.

**Comment by iglocska** [comment]
Besides the exclusions - not really sure. Did you identify a specific source for the massive correlations? Such as data coming from a specific feed for example.

**Comment by jloehel** [comment]
> Besides the exclusions - not really sure. Did you identify a specific source for the massive correlations? Such as data coming from a specific feed for example.

Yes, there are a few feeds which overlap. Some between 60 and 80%. Is it possible to exclude correlations just between specific events/feeds? In example between `Tor Exit Nodes`/`Tor All Nodes` or `Phishtank`/`Phishscore`/`OpenPhish`. Properly the MISP feeds from abuse.ch and osint.digitalside.it cause a lot of correlations. Especially osint.digitalside.it produces a lot of small events. I deactivated the feed because abuse.ch offers now its own MISP feeds. Is it enough to delete the problematic events?

**Comment by Tommi852** [comment]
I disabled Abuse.ch MISP feeds because they seemed to generate around 100 correlations per event. I had all the Abuse.ch feeds, Bazaar, Threatfox, and URLhaus.
I deleted all abuse.ch events which lowered my attributes from 1,300,000 to 400,000.
I also disabled correlations to purge them, but have yet to rebuild them without the Abuse.ch data.
**Update:** After rebuilding the correlations I have just 18513 correlations. So the Abuse.ch MISP feeds were the culprit.

I also noticed from Top Correlations Index that the most correlated value was Application/x-dosexec which caused 355,543 correlations. There were also other application/* values which created quite a lot of correlations. Those values came from events which had mime-types. I don't see any value in correlation the mime-types, so could it be possible to add a function to create correlation exclusion rules based on different categories/types?

**Comment by Tommi852** [comment]
After some testing it seems like only the URLhaus MISP feed by Abuse.ch is creating the excessive correlations.
I believe the overlapping attributes are only overlapping within the URLhaus feed, because feed overlap analysis matrix didn't show any significant overlapping with other feeds.
ThreatFox MISP feed works great and doesn't create excessive correlations.

MISP is once again working as it is supposed to. Thank you for your answers.

**Comment by iglocska** [comment]
Awesome, thanks for the feedback!
